src/angles.jl

```diff
@@ -8,32 +8,53 @@
 Angle ∠ABC between rays BA and BC.
 See https://en.wikipedia.org/wiki/Angle.
 
-Uses the two-argument form of `atan` returning value in range [-π, π].
-See https://en.wikipedia.org/wiki/Atan2.
+Return a value in range [-π, π] for 2D
+angles and a value in range [0, π] for
+3D angles.
 
 ## Examples
 
 ```julia
 ∠(Point(1,0), Point(0,0), Point(0,1)) == π/2
 ```
 """
-∠(A::P, B::P, C::P) where {P<:Point{2}} = ∠(A-B, C-B)
-∠(A::P, B::P, C::P) where {P<:Point{3}} = ∠(A-B, C-B)
+∠(A::Point, B::Point, C::Point) = ∠(A-B, C-B)
 
 """
     ∠(u, v)
 
 Angle between vectors u and v.
 See https://en.wikipedia.org/wiki/Angle.
 
-Uses the two-argument form of `atan` returning value in range [-π, π].
-See https://en.wikipedia.org/wiki/Atan2.
+Return a value in range [-π, π] for 2D
+angles and a value in range [0, π] for
+3D angles.
 
 ## Examples
 
 ```julia
 ∠(Vec(1,0), Vec(0,1)) == π/2
 ```
 """
-∠(u::V, v::V) where {V<:Vec{2}} = atan(u × v, u ⋅ v)  # preserve sign
-∠(u::V, v::V) where {V<:Vec{3}} = atan(norm(u × v), u ⋅ v)  # discard sign
+function ∠(u::Vec, v::Vec)
+  T = eltype(u)
+
+  uunit = unitize(u)
+  vunit = unitize(v)
+
+  y = norm2(uunit .- vunit)
+  x = norm2(uunit .+ vunit)
+
+  a = 2 * atan(y, x)
+
+  a = !(signbit(a) || signbit(pi - a)) ? a : (signbit(a) ? zero(T) : (T)(pi))
+
+  ifelse(isnegangle(uunit, vunit), -a, a)
+end
+
+@inline norm2(u::Vec) = sqrt(foldl(+, abs2.(u)))
+
+@inline unitize(u::Vec) = u ./ norm2(u)
```

Comment:
Given that our `Vec` is a `StaticVector` from StaticArrays.jl I think we can use their implementation of LinearAlgebra.normalize and LinearAlgebra.norm here.